builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -163,17 +169,20 @@ type puller struct {
 	resolveErr       error
 	resolver         remotes.Resolver
 	config           []byte
+	platform         ocispec.Platform
 }
 
-func (p *puller) mainManifestKey(dgst digest.Digest) (digest.Digest, error) {
+func (p *puller) mainManifestKey(dgst digest.Digest, platform ocispec.Platform) (digest.Digest, error) {
 	dt, err := json.Marshal(struct {
-		Digest digest.Digest
-		OS     string
-		Arch   string
+		Digest  digest.Digest
+		OS      string
+		Arch    string
+		Variant string `json:",omitempty"`
```

Comment:
Do we need the OSversion + features in these structs as well, or can those be added at a later stage if needed?

Reply:
We don't as the containerd puller doesn't use these fields. Should they be needed in the future they can be added in backward compatible manner.

Reply:
Alright, thanks for checking!